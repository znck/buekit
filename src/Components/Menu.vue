<script>
import { cssModule, mergeData, addStyleToVnode } from '../utils'
import { tag } from '../mixins'

export default {
  name: 'Menu',
  functional: true,
  mixins: [tag],
  props: { tag: { default: 'aside' } },
  render (h, ctx) {
    const { tag } = ctx.props

    return addStyleToVnode(
      h(tag, ctx.data, ctx.children), ctx.$style['menu']
    )
  }
}
</script>

<style lang="scss" module>
@import '~bulma/sass/utilities/_all.sass';
@import '../_base.scss';
@import '~bulma/sass/components/menu.sass';
</style>
