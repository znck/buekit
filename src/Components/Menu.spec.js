import { shallow } from 'vue-test-utils'
import { createRenderer } from 'vue-server-renderer'

import Menu from './Menu.vue'

describe('Menu.vue', () => {

  it ('matches snapshot', () => {
    const wrapper = shallow(Menu, {})
    const renderer = createRenderer()

    renderer.renderToString(wrapper.vm, (err, str) => {
      if (err) throw new Error(err)
      expect(str).toMatchSnapshot()
    })
  })
})
