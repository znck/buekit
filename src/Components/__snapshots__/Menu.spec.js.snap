// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu.vue matches snapshot 1`] = `<aside class="menu"></aside>`;
